packages/sit-onyx/src/components/OnyxDataGrid/features/resizing/resizing.ts

```diff
@@ -0,0 +1,97 @@
+import { h, ref, type Slot } from "vue";
+import { createFeature, type InternalColumnConfig } from "..";
+import type { DataGridEntry } from "../../types";
+import "./resizing.scss";
+import type { ResizingOptions } from "./types";
+
+export const RESIZING_FEATURE = Symbol("Resizing");
+export const useResizing = createFeature(
+  <TEntry extends DataGridEntry>(options?: ResizingOptions<TEntry>) => {
+    let column: Readonly<InternalColumnConfig<TEntry>>;
+    const min = 70;
+    const headerBeingResized = ref<Record<PropertyKey, HTMLElement>>({});
+
+    const onMouseMove = (ev: { clientX: number }) =>
+      requestAnimationFrame(() => {
+        // Calculate the desired width
+        const horizontalScrollOffset = document.documentElement.scrollLeft;
+        const width =
+          horizontalScrollOffset + ev.clientX - headerBeingResized.value[column.key].offsetLeft;
+
+        headerBeingResized.value[column.key].style.width = Math.max(min, width) + "px";
+      });
+
+    // Clean up event listeners, classes, etc.
+    const onMouseUp = () => {
+      window.removeEventListener("mousemove", onMouseMove);
+      window.removeEventListener("mouseup", onMouseUp);
+      headerBeingResized.value[column.key].classList.remove("header--being-resized");
+      headerBeingResized.value = {};
+    };
+
+    const initResize = (ev: Event, cols: Readonly<InternalColumnConfig<TEntry>>) => {
+      const target = ev.target as HTMLElement;
+      column = cols;
+
+      headerBeingResized.value[column.key] = target.closest("th")!;
+      window.addEventListener("mousemove", onMouseMove);
+      window.addEventListener("mouseup", onMouseUp);
+      headerBeingResized.value[column.key].classList.add("header--being-resized");
+    };
+
+    const modifyColumns = (cols: Readonly<InternalColumnConfig<TEntry>[]>) => {
+      if (!options?.columnResizing) {
+        return [...cols];
+      }
+
+      return cols.map((column) => {
+        return {
+          key: column.key,
+          type: column.key,
+          tdAttributes: {
+            class:
+              headerBeingResized.value !== null &&
+              Object.keys(headerBeingResized.value).includes(column.key as string)
+                ? "resize-border"
+                : "",
+          },
+        };
```

Comment:
We have to be careful with not overwriting other column configuration, so we should merge as careful as possible:
```suggestion
        const tdAttributes = {
            class:
                headerBeingResized.value !== null &&
                Object.keys(headerBeingResized.value).includes(column.key as string)
                ? "resize-border"
                : "",
        };
        return {
          ...column,
          tdAttributes: mergeVueProps(tdAttributes, column.tdAttributes),
        };
```